librocksdb_sys/crocksdb/c.cc

```diff
@@ -2915,4 +2927,341 @@ const char* crocksdb_pinnableslice_value(const crocksdb_pinnableslice_t* v,
+    return rep->filter_policy_name.data();
+  case COMPARATOR_NAME:
+    if (slen) *slen = rep->comparator_name.size();
+    return rep->comparator_name.data();
+  case MERGE_OPERATOR_NAME:
+    if (slen) *slen = rep->merge_operator_name.size();
+    return rep->merge_operator_name.data();
+  case PREFIX_EXTRACTOR_NAME:
+    if (slen) *slen = rep->prefix_extractor_name.size();
+    return rep->prefix_extractor_name.data();
+  case PROPERTY_COLLECTORS_NAMES:
+    if (slen) *slen = rep->property_collectors_names.size();
+    return rep->property_collectors_names.data();
+  case COMPRESSION_NAME:
+    if (slen) *slen = rep->compression_name.size();
+    return rep->compression_name.data();
+  }
+  return "";
+}
+
+crocksdb_user_collected_properties_t*
+crocksdb_table_properties_get_user_properties(crocksdb_table_properties_t* props) {
+  props->user_props_.rep_ =
+    const_cast<UserCollectedProperties*>(&props->rep_->user_collected_properties);
+  return &props->user_props_;
+}
+
+/* Table Properties Collection */
+
+struct crocksdb_table_properties_collection_t {
+  TablePropertiesCollection* rep_ = nullptr;
+
+  crocksdb_table_properties_collection_t() {
+    rep_ = new TablePropertiesCollection;
+  }
+
+  ~crocksdb_table_properties_collection_t() {
+    delete rep_;
+  }
+};
+
+struct crocksdb_table_properties_collection_iterator_t {
+  const TablePropertiesCollection* rep_ = nullptr;
+  TablePropertiesCollection::const_iterator iter_;
+  crocksdb_table_properties_t props_;
+};
+
+crocksdb_table_properties_collection_t*
+crocksdb_table_properties_collection_create() {
+  return new crocksdb_table_properties_collection_t;
+}
+
+void crocksdb_table_properties_collection_destroy(
+  crocksdb_table_properties_collection_t* collection) {
+  delete collection;
+}
+
+crocksdb_table_properties_collection_iterator_t*
+crocksdb_table_properties_collection_iter_create(crocksdb_table_properties_collection_t* c) {
+  auto it = new crocksdb_table_properties_collection_iterator_t;
+  it->rep_ = c->rep_;
+  it->iter_ = c->rep_->begin();
+  return it;
+}
+
+void crocksdb_table_properties_collection_iter_destroy(
+  crocksdb_table_properties_collection_iterator_t* it) {
+  delete it;
+}
+
+unsigned char crocksdb_table_properties_collection_iter_valid(
+  crocksdb_table_properties_collection_iterator_t* it) {
+  return it->iter_ != it->rep_->end();
+}
+
+void crocksdb_table_properties_collection_iter_next(
+  crocksdb_table_properties_collection_iterator_t* it) {
+  (it->iter_)++;
+}
+
+const char* crocksdb_table_properties_collection_iter_key(
+  crocksdb_table_properties_collection_iterator_t* it, size_t* klen) {
+  if (klen) {
+    *klen = it->iter_->first.size();
+  }
+  return it->iter_->first.data();
+}
+
+crocksdb_table_properties_t* crocksdb_table_properties_collection_iter_value(
+  crocksdb_table_properties_collection_iterator_t* it) {
+  it->props_.rep_ = const_cast<TableProperties*>(it->iter_->second.get());
```

Comment:
Yep, but there is a place I can't avoid, because rocksdb returns a `const TableProperties*`, maybe I just cast that one.